BDArmory/Settings/RWPSettings.cs

```diff
@@ -145,6 +153,24 @@ public static void RestoreSettings(bool temp = false)
       if (BDArmorySettings.DEBUG_OTHER) Debug.Log($"[BDArmory.RWPSettings]: Restored settings: " + string.Join(", ", settings.Select(kvp => $"{kvp.Key}={kvp.Value}")));
     }
 
+        /// <summary>
+        /// set whether or not a BDAsetting should be filtered by the store/restore setting
+        /// </summary>
+        public static void SetRWPFilters()
+        {
+            if (!RWPOverrides.ContainsKey(0)) return;
+            Debug.Log($"[BDArmory.RWPSettings]: Setting RWP setting filters");
+            var fields = typeof(BDArmorySettings).GetFields(BindingFlags.Public | BindingFlags.Static | BindingFlags.DeclaredOnly);
+            foreach (var field in fields)
+            {
+                if (field == null) continue;
+                if (!field.IsDefined(typeof(BDAPersistentSettingsField), false)) continue;
+                BDAPersistentSettingsField attr = field.GetCustomAttribute<BDAPersistentSettingsField>();
+                attr.RWPFilter = (RWPOverrides[0].Keys.Contains(field.Name) || RWPOverrides[BDArmorySettings.RUNWAY_PROJECT_ROUND].Keys.Contains(field.Name));
```

Comment:
It seems that setting the value like this doesn't work. If you get the `attr` again and read the `attr.RWPFilter` value, it's reverted to `false`.

We can achieve the same thing though by simply using a HashSet to track what we're currently modifying. E.g.,
```diff
diff --git a/BDArmory/Settings/BDAPersistentSettingsField.cs b/BDArmory/Settings/BDAPersistentSettingsField.cs
index edbd62b85..d2e8efa0c 100644
--- a/BDArmory/Settings/BDAPersistentSettingsField.cs
+++ b/BDArmory/Settings/BDAPersistentSettingsField.cs
@@ -10,7 +10,6 @@ namespace BDArmory.Settings
     [AttributeUsage(AttributeTargets.Field)]
     public class BDAPersistentSettingsField : Attribute
     {
-        public bool RWPFilter = false;
         public BDAPersistentSettingsField()
         {
         }
diff --git a/BDArmory/Settings/RWPSettings.cs b/BDArmory/Settings/RWPSettings.cs
index a929867d2..e166f8cd7 100644
--- a/BDArmory/Settings/RWPSettings.cs
+++ b/BDArmory/Settings/RWPSettings.cs
@@ -88,16 +88,17 @@ public class RWPSettings
       }}
     };
     public static Dictionary<int, int> RWPRoundToIndex = new() { { 0, 0 } }, RWPIndexToRound = new() { { 0, 0 } }; // Helpers for the UI slider.
+    static readonly HashSet<string> currentFilter = [];
 
     public static void SetRWP(bool enabled, int round)
     {
       if (enabled)
       {
-        SetRWPFilters();
-        if (!RWPEnabled) StoreSettings(); // Enabling RWP. Store the non-RWP settings.
-        else RestoreSettings(); // Was previously enabled. Restore the non-RWP settings before applying new ones.
+        if (RWPEnabled) RestoreSettings(); // Revert to the original settings if we've been modifying things.
+        SetRWPFilters(); // Figure out what we're going to change and tag those settings.
+        StoreSettings(); // Store the original values of the settings that we're going to change.
         SetOverrides(0); // Set global RWP settings.
-        SetOverrides(round); // Set the round-specific settings.
+        if (round > 0) SetOverrides(round); // Set the round-specific settings.
       }
       else if (!enabled && RWPEnabled) // Disabling RWP.
       {
@@ -110,7 +111,7 @@ public static void SetRWP(bool enabled, int round)
     static Dictionary<string, object> tempSettings = []; // Temporary settings for shuffling things around.
 
     /// <summary>
-    /// Store the non-RWP settings.
+    /// Store the settings that RWP is going to change.
     /// </summary>
     public static void StoreSettings(bool temp = false)
     {
@@ -121,14 +122,7 @@ public static void StoreSettings(bool temp = false)
       var fields = typeof(BDArmorySettings).GetFields(BindingFlags.Public | BindingFlags.Static | BindingFlags.DeclaredOnly);
       foreach (var field in fields)
       {
-                //if (field.Name.EndsWith("_SETTINGS_TOGGLE")) continue; // Don't toggle the section headers.
-                //if (field.Name.StartsWith("RUNWAY_PROJECT")) continue; // Skip settings beginning with RWP (toggle and slider) to avoid recursion.
-                //if (field.Name.StartsWith("VESSEL_SPAWN_")) continue; // Skip spawn settings so they are the same between RWP and non-RWP.
-                //if (field.Name.StartsWith("TOURNAMENT_")) continue; // Skip tournament settings so they are the same between RWP and non-RWP.
-                if (!field.IsDefined(typeof(BDAPersistentSettingsField), false)) continue;
-
-                BDAPersistentSettingsField attr = field.GetCustomAttribute<BDAPersistentSettingsField>();
-                if (!attr.RWPFilter)
+        if (!currentFilter.Contains(field.Name))
         {
           // Debug.Log($"[BDArmory.RWPSettings]: {field.Name} is non-RWP setting, skipping...");
           continue;
@@ -136,6 +130,7 @@ public static void StoreSettings(bool temp = false)
 
         settings.Add(field.Name, field.GetValue(null));
       }
+      if (BDArmorySettings.DEBUG_OTHER) Debug.Log($"[BDArmory.RWPSettings]: Stored settings: " + string.Join(", ", settings.Select(kvp => $"{kvp.Key}={kvp.Value}")));
     }
 
     /// <summary>
@@ -143,8 +138,8 @@ public static void StoreSettings(bool temp = false)
     /// </summary>
     public static void RestoreSettings(bool temp = false)
     {
-      Debug.Log($"[BDArmory.RWPSettings]: Restoring {(temp ? "temporary" : "base")} settings.");
       var settings = temp ? tempSettings : storedSettings;
+      Debug.Log($"[BDArmory.RWPSettings]: Restoring {(temp ? "temporary" : "base")} settings.");
       foreach (var setting in settings.Keys)
       {
         var field = typeof(BDArmorySettings).GetField(setting, BindingFlags.Public | BindingFlags.Static | BindingFlags.DeclaredOnly);
@@ -161,14 +156,14 @@ public static void SetRWPFilters()
     {
       if (!RWPOverrides.ContainsKey(0)) return;
       Debug.Log($"[BDArmory.RWPSettings]: Setting RWP setting filters");
+      currentFilter.Clear();
       var fields = typeof(BDArmorySettings).GetFields(BindingFlags.Public | BindingFlags.Static | BindingFlags.DeclaredOnly);
       foreach (var field in fields)
       {
         if (field == null) continue;
         if (!field.IsDefined(typeof(BDAPersistentSettingsField), false)) continue;
-                BDAPersistentSettingsField attr = field.GetCustomAttribute<BDAPersistentSettingsField>();
-                attr.RWPFilter = (RWPOverrides[0].Keys.Contains(field.Name) || RWPOverrides[BDArmorySettings.RUNWAY_PROJECT_ROUND].Keys.Contains(field.Name));
-                if (attr.RWPFilter) Debug.Log($"[BDArmory.RWPSettings]: {field.Name} is RWP setting, tagging...");
+        if (RWPOverrides[0].Keys.Contains(field.Name) || RWPOverrides[BDArmorySettings.RUNWAY_PROJECT_ROUND].Keys.Contains(field.Name))
+          currentFilter.Add(field.Name);
       }
     }
 
```

Reply:
yeah, if we're no longer doing [BDAPersistentSettingsField(RWPFilter = true)] for filtering, then simply having a hashset and keeping everything local would be much simpler.